### PR TITLE
Naming: Float suffixes 

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
**What this PR does / why we need it**:

When generating hints for naming of nested anonymous types, float the suffixes to the end.

**Special notes for your reviewer**:

Most of the changes are in `name_types_for_crd` and the rest flows from that.

**How does this PR make you feel**:
gif

### CODE REVIEW

File: v2/tools/generator/internal/astmodel/type_name.go [hunk 1]
```diff
@@ -227,6 +227,8 @@ const (
 	SpecSuffix = "_Spec"
 	// StatusSuffix is the suffix used for all Status types
 	StatusSuffix = "_STATUS"
+	// ArmSuffix is the suffix used for all ARM types
+	ArmSuffix = "_ARM"
```
Comment (matthchr) [inline v2/tools/generator/internal/astmodel/type_name.go:231]: minor: I think `ARm` should be `ARM` here, with our styling. See for example `pipeline.CreateARMTypes` or `MakeSubscriptionScopeARMID`

**Review by matthchr — APPROVED** [review]

File: v2/azure-arm.yaml [hunk 2]
```diff
@@ -51,89 +51,118 @@ typeFilters:
...
     group: machinelearningservices
     name: Workspaces_Service*
     because: Some types use OneOf to model inheritance, which we don't currently support.
+
   - action: prune
     group: resources
     name: DeploymentScript*
     because: Some types use OneOf to model inheritance, which we don't currently support.
+
+  - action: prune
```
[the remark this answers is not in the document]
Comment (theunrepentantgeek) [reply v2/azure-arm.yaml:147]: Not really sure, but the Swagger branch has the same issue so I figured it was something we could punt for later.

File: v2/tools/generator/internal/astmodel/type_definition.go [hunk 3]
```diff
@@ -7,11 +7,9 @@ package astmodel
 
 import (
 	"fmt"
-	"go/token"
-	"strings"
-
 	"github.com/dave/dst"
 	"github.com/pkg/errors"
+	"go/token"
```
[the remark this answers is not in the document]
Comment (theunrepentantgeek) [reply v2/tools/generator/internal/astmodel/type_definition.go:12]: Fixed.

File: v2/tools/generator/internal/astmodel/type_name.go (hunk 1 above)
Comment (theunrepentantgeek) [reply v2/tools/generator/internal/astmodel/type_name.go:231]: Fixed.

**Pull request merged**